### PR TITLE
Adding exec.SelectOne

### PR DESCRIPTION
**Pull request by pascallouisperez**
Convenience method to select a single row into a struct, similar to exec.Get but generalized to allow a query (with arguments).

cc @petermattis, @jbowens

### CODE REVIEW

File: db.go [hunk 1]
```diff
@@ -562,6 +562,20 @@ func (db *DB) Select(dest interface{}, q interface{}, args ...interface{}) error
...
+// It is ok for dest to refer to a struct that has not been bound to a
+// table. This allows querying for values that return transient
+// columnts. For example, "SELECT count(*) ..." will return a "count"
+// column.
+//
+// dest must be a pointer to a struct.
+func (db *DB) SelectOne(dest interface{}, query interface{}, args ...interface{}) error {
+	return getObjectByQuery(db, db, dest, query, args...)
+}
+
```
Comment (petermattis) [inline db.go]: I think similar functionality can be had from `QueryRow(...).Scan(&val)`. I'm not sure it is worthwhile to support both methods and the `QueryRow` version has the advantage of matching up with the `database/sql` interfaces.

Comment (pascallouisperez) [inline db.go]: Unfortunately not, you'll need to align the values. In the test, if you replace the line

    if err := db.SelectOne(&user, "SELECT * FROM users WHERE name = ?", "one"); err != nil {

by

    if err := db.QueryRow("SELECT * FROM users WHERE name = ?", "one").Scan(&user); err != nil {

you'll get

    	db_test.go:1016: sql: expected 3 destination arguments in Scan, not 1

Comment (petermattis) [inline db.go]: Oops. You're right that `Scan` won't work, but [StructScan](http://godoc.org/github.com/square/squalor#Row.StructScan) should: `QueryRow(...).StructScan(&val)`

Comment (pascallouisperez) [inline db.go]: Yup, good point. Personally not a fan of the asymmetry, but it works.

**Pull request closed without merging**